src/figuresClasses.ts

```diff
@@ -1,11 +1,84 @@
-export interface Figure {}
+type Shape = `triangle` | `circle` | `rectangle`;
+type Color = `red` | `green` | `blue`;
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('your error message');
```

Comment:
Why u don't explain the error?
Do it in all such cases!
```suggestion
      throw new Error('...........');
```